### PR TITLE
583

### PR DESCRIPTION
**Pull request by FineAndDandy**

### CODE REVIEW

**Comment by FineAndDandy** [comment]
this was built on #575 since they are highly related, and 575 is approved, but not yet merged

File: warehouse/query-core/src/test/java/datawave/query/index/lookup/IntersectionTest.java [hunk 1]
```diff
@@ -209,4 +214,458 @@ public void testIntersection_EmptyExceededValueThreshold() throws ParseException
...
+        IndexInfo ii1;
+        if (uids != null) {
+            ii1 = new IndexInfo(uids);
+        } else {
+            ii1 = new IndexInfo(-1);
+        }
+        JexlNode n1 = JexlNodeFactory.buildEQNode(field, value);
+        ii1.applyNode(n1);
+        Iterator<Tuple2<String,IndexInfo>> i1 = Arrays.asList(new Tuple2<>(shard, ii1)).iterator();
+        return ScannerStream.variable(i1, n1);
```
Comment (apmoriarty) [inline warehouse/query-core/src/test/java/datawave/query/index/lookup/IntersectionTest.java]: Won't hold up an approval, but I would rather just see the variable names "indexInfo" and "node"

File: warehouse/query-core/src/main/java/datawave/query/index/lookup/IndexInfo.java [hunk 2]
```diff
@@ -458,67 +469,52 @@ protected IndexInfo intersect(long maxPossibilities, Iterable<IndexMatch> matchI
...
-                allNodes.add(delayedNode);
-            else
-                allNodes.add(nodeColl.iterator().next());
+            
+            // it is counter intuitive that this is an OR, but since each indexMatch is actually a potential different query path an or is appropriate here
+            // example (A || B) && C
+            // IndexMatch - A == 'a'
+            // IndexMatch - B == 'b'
+            // IndexMatch - C == infinite
+            // the merge node's matches actually represent (A && C) || (A && B)
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/index/lookup/IndexInfo.java]: did you mean to say (B && C)

File: warehouse/query-core/src/main/java/datawave/query/index/lookup/IndexInfo.java [hunk 3]
```diff
@@ -415,6 +417,7 @@ public boolean intersect(List<JexlNode> delayedNodes) {
             nodeSet.add(match.getNode());
             
             match.set(TreeFlatteningRebuildingVisitor.flatten(JexlNodeFactory.createAndNode(nodeSet)));
+            // TODO this may need to be of type AND for nested logic to be correct
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/index/lookup/IndexInfo.java]: Not understanding this TODO.....

File: warehouse/query-core/src/main/java/datawave/query/index/lookup/IndexInfo.java (hunk 2 above)
Comment (ivakegg) [reply warehouse/query-core/src/main/java/datawave/query/index/lookup/IndexInfo.java]: instead of (A && B) ?

**Pull request merged**